### DISCUSSION TITLE
Is there any way to extract coherence scores from already saved BERTopic model?

### DISCUSSION THREAD

**Discussion by QuantityIntel** [MaartenGr/BERTopic#900] [Q&A]
I want to compare the coherence scores of the models I trained to select an optimal number of topics.

`bertopic  = BERTopic()`
`mod = bertopic.load(myModel)`

However, `mod.__dict__` returns the only the folloiwng information:

![image](https://user-images.githubusercontent.com/46237445/210038564-af07aae1-c590-490d-9263-b3ac22def295.png)

Can I get a coherence score from the model?

**Comment by MaartenGr** [comment]
It is not possible to extract the coherence score from BERTopic as there is no single best way to perform evaluation in topic modeling. Instead, I would advise using [OCTIS](https://github.com/MIND-Lab/OCTIS) for evaluating your topic model.
